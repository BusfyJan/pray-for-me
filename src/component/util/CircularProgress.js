import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import { CircularProgress as MaterialCircularProgress } from 'material-ui/Progress'
import { blue } from 'material-ui/colors'

const MaterialCircularProgressStyled = styled(MaterialCircularProgress)`
  margin-right: 0px;
  color: ${props => (props.color ? props.color : blue[800])} !important;
`

const CircularProgress = ({ color, size, thickness }) => (
  <MaterialCircularProgressStyled
    color={color}
    size={size}
    thickness={thickness}
  />
)

CircularProgress.propTypes = {
  color: PropTypes.string,
  size: PropTypes.number.isRequired,
  thickness: PropTypes.number.isRequired
}

CircularProgress.defaultProps = {
  color: blue[800]
}
export default CircularProgress
