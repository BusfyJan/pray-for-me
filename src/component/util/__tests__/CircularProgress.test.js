import React from 'react'
import CircularProgress from '../CircularProgress'
import renderer from 'react-test-renderer'

test('CircularProgress render', () => {
  const component = renderer.create(
    <CircularProgress size={20} thickness={5} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
