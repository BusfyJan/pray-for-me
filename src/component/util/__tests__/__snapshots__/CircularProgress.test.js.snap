// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularProgress render 1`] = `
<div
  className="MuiCircularProgress-root-1 sc-bdVaJa gFCFAM"
  role="progressbar"
  style={
    Object {
      "height": 20,
      "width": 20,
    }
  }
>
  <svg
    className="MuiCircularProgress-svgIndeterminate-4"
    viewBox="0 0 50 50"
  >
    <circle
      className="MuiCircularProgress-circle-6 MuiCircularProgress-circleIndeterminate-7"
      cx={25}
      cy={25}
      fill="none"
      r={20}
      strokeWidth={5}
      style={Object {}}
    />
  </svg>
</div>
`;
